src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -127,6 +128,7 @@ export class TerminalService implements ITerminalService {
 			terminalInstance.setVisible(i === terminalIndex);
 		});
 		this._onActiveInstanceChanged.fire();
+		this.getActiveInstance().focus(true);
```

Comment:
I believe the reason focus is not run in `setActiveInstanceByIndex` is because on launch we do not want the terminal to be focused when the workbench is restored, instead the editor needs to be focused.

Could we isolate this particular case? When a terminal exits AND it is not the last terminal AND the terminal was focused previously. The last case prevents the terminal hijacking focus from the editor if the shell process dies for example.